src/main/java/de/tum/in/www1/artemis/domain/enumeration/ExerciseLifecycle.java

```diff
@@ -13,27 +15,47 @@ public enum ExerciseLifecycle implements IExerciseLifecycle {
         public ZonedDateTime getDateFromExercise(Exercise exercise) {
             return exercise.getReleaseDate();
         }
+
+        @Override
+        public ZonedDateTime getDateFromQuizBatch(QuizBatch quizBatch, QuizExercise quizExercise) {
+            return getDateFromExercise(quizExercise);
+        }
     },
     START {
 
         @Override
         public ZonedDateTime getDateFromExercise(Exercise exercise) {
             return exercise.getStartDate();
         }
+
+        @Override
+        public ZonedDateTime getDateFromQuizBatch(QuizBatch quizBatch, QuizExercise quizExercise) {
+            return quizBatch.getStartTime();
+        }
     },
     DUE {
 
         @Override
         public ZonedDateTime getDateFromExercise(Exercise exercise) {
             return exercise.getDueDate();
         }
+
+        @Override
+        public ZonedDateTime getDateFromQuizBatch(QuizBatch quizBatch, QuizExercise quizExercise) {
+            return quizExercise.getDueDate();
+        }
     },
     ASSESSMENT_DUE {
 
         @Override
         public ZonedDateTime getDateFromExercise(Exercise exercise) {
             return exercise.getAssessmentDueDate();
         }
+
+        @Override
+        public ZonedDateTime getDateFromQuizBatch(QuizBatch quizBatch, QuizExercise quizExercise) {
+            throw new IllegalStateException("Unexpected QuizExercise Lifecycle State " + this + " for exercise: " + quizExercise);
+        }
```

Comment:
It could be a good idea to extract the method or error message to a constant to avoid repetetive code and errors in the future.